aim/ext/task_queue/queue.py

```diff
@@ -65,3 +67,74 @@ def stop_workers(self):
 
     def __del__(self):
         self.stop_workers()
+
+    def _try_worker(self, task_f, *args, **kwargs):
+        task_f(*args, **kwargs)
+
+    def _put(self, task_f, *args, **kwargs):
+        warn_queue_full = False
+        backlog_size = self._queue.qsize()
+        if backlog_size > self.max_backlog * 0.8:  # queue is 80% full
+            warn_queue_full = True
+        self._queue.put((task_f, args, kwargs))
+        return warn_queue_full
+
+
+class TaskQueueWithRetry(TaskQueue):
+    def __init__(self, name, num_workers=1, max_queue_memory=0, register_at_exit=True,
+                 retry_count=0, retry_interval=0):
+        super().__init__(name, num_workers, register_at_exit)
+
+        self.retry_count = retry_count or 1
+        self.retry_interval = retry_interval
+
+        self.max_memory_usage = max_queue_memory
+        self.current_memory_usage = 0
+
+    def _try_worker(self, task_f, *args, **kwargs):
+        assert len(kwargs) == 0
```

Comment:
is this required for the queue size calculation?

Reply:
yes, after refactoring I removed `**kwarg` param with assertion.